src/components/ha-icon.ts

```diff
@@ -161,7 +160,7 @@ export class HaIcon extends LitElement {
       return nothing;
     }
     if (this._legacy) {
-      return html`<iron-icon .icon=${this.icon}></iron-icon>`;
+      return nothing;
```

Comment:
I don't think anything actually uses this, but if it is used, we could revert this, but let the user supply the `iron-icon` element.